include/z64.h

```diff
@@ -969,6 +969,77 @@ typedef struct {
     /* 0x4C */ u32 unk_4C;
 } PreRender; // size = 0x50
 
+#define TRANS_TRIGGER_OUT -20 // transition is going out
+#define TRANS_TRIGGER_OFF 0 // transition is not active
+#define TRANS_TRIGGER_IN 20 // activate transition to come in
+
+typedef enum {
+    /* 00 */ TRANS_MODE_OFF,
+    /* 01 */ TRANS_MODE_SETUP,
+    /* 02 */ TRANS_MODE_INSTANCE_INIT,
+    /* 03 */ TRANS_MODE_INSTANCE_RUNNING,
+    /* 04 */ TRANS_MODE_FILL_WHITE_INIT,
+    /* 05 */ TRANS_MODE_FILL_IN,
+    /* 06 */ TRANS_MODE_FILL_OUT,
+    /* 07 */ TRANS_MODE_FILL_BROWN_INIT,
+    /* 08 */ TRANS_MODE_08, // unused
+    /* 09 */ TRANS_MODE_09, // unused
+    /* 10 */ TRANS_MODE_INSTANT,
+    /* 11 */ TRANS_MODE_INSTANCE_WAIT,
+    /* 12 */ TRANS_MODE_SANDSTORM_INIT,
+    /* 13 */ TRANS_MODE_SANDSTORM,
+    /* 14 */ TRANS_MODE_SANDSTORM_END_INIT,
+    /* 15 */ TRANS_MODE_SANDSTORM_END,
+    /* 16 */ TRANS_MODE_CS_BLACK_FILL_INIT,
+    /* 17 */ TRANS_MODE_CS_BLACK_FILL
+} TransitionMode;
+
+typedef enum {
+    /* 00 */ TRANS_TYPE_WIPE,
+    /* 01 */ TRANS_TYPE_TRIFORCE,
+    /* 02 */ TRANS_TYPE_FADE_BLACK,
+    /* 03 */ TRANS_TYPE_FADE_WHITE,
+    /* 04 */ TRANS_TYPE_FADE_BLACK_FAST,
+    /* 05 */ TRANS_TYPE_FADE_WHITE_FAST,
+    /* 06 */ TRANS_TYPE_FADE_BLACK_SLOW,
+    /* 07 */ TRANS_TYPE_FADE_WHITE_SLOW,
+    /* 08 */ TRANS_TYPE_WIPE_FAST,
+    /* 09 */ TRANS_TYPE_FILL_WHITE2, 
+    /* 10 */ TRANS_TYPE_FILL_WHITE,
+    /* 11 */ TRANS_TYPE_INSTANT,
+    /* 12 */ TRANS_TYPE_FILL_BROWN,
+    /* 13 */ TRANS_TYPE_FADE_WHITE_CS_DELAYED,
+    /* 14 */ TRANS_TYPE_SANDSTORM_PERSIST,
+    /* 15 */ TRANS_TYPE_SANDSTORM_END,
+    /* 16 */ TRANS_TYPE_CS_BLACK_FILL,
+    /* 17 */ TRANS_TYPE_FADE_WHITE_INSTANT,
+    /* 18 */ TRANS_TYPE_FADE_GREEN,
+    /* 19 */ TRANS_TYPE_FADE_BLUE,
+    // transition types 20 - 31 are unused
+    // transition types 32 - 55 are constructed using the TRANS_TYPE_CIRCLE macro
+    /* 56 */ TRANS_TYPE_MAX = 56
+} TransitionType;
+
+typedef enum {
+    /* 0 */ TC_NORMAL,
+    /* 1 */ TC_WAVE,
+    /* 2 */ TC_RIPPLE,
+    /* 3 */ TC_STARBURST
+} TransitionCircleType;
+
+typedef enum {
+    /* 0 */ TC_BLACK,
+    /* 1 */ TC_WHITE,
+    /* 1 */ TC_GRAY
+} TransitionCircleColor;
+
+typedef enum {
+    /* 0 */ TC_FAST,
+    /* 1 */ TC_SLOW
+} TransitionCircleSpeed;
+
+#define TRANS_TYPE_CIRCLE(type, color, speed, unk_14) (((unk_14 & 1) << 5) | ((color & 3) << 3) | ((type & 3) << 1) | (speed & 1))
```

Comment:
where does unk_14 come from? it just looks like what makes the values start at 32 (it's always 1 right?)

Reply:
it is a property of the circle which I was unable to determine its purpose.
It does make sense that it makes them start at 32 though, in which case the code in the circles that depends on this value being 0 is probably dead code

Reply:
0 gets used in practice when leaving an area (transitioning in), so im clearly missing something. will look into it more

Reply:
i see whats happening now and im mad at how dumb it is. 
for context, the relevant code in the circle happens in its setType callback, which gets called in z_play via `globalCtx->transitionCtx.setType()`

if a transition type is 32 or above, it will pass its type as the arg to the function 
```
if ((globalCtx->transitionCtx.transitionType >> 5) == 1) {
    globalCtx->transitionCtx.setType(&globalCtx->transitionCtx.data,
                                     globalCtx->transitionCtx.transitionType | 0x80);
}
```
this is where unk_14 would gets its value from the type. but as you mentioned, it has to be 1 anyway for this to be true.

Now, later on in z_play it calls the setType callback *again* (this time for all the other types that are below 32)
```
if (globalCtx->transitionTrigger == TRANS_TRIGGER_OUT) {
    globalCtx->transitionCtx.setType(&globalCtx->transitionCtx.data, 1);
} else {
    globalCtx->transitionCtx.setType(&globalCtx->transitionCtx.data, 2);
}
```
since it is not passing 0x80 to the circle type, it does not do the normal circle setup. if TRIGGER_IN (leaving), unk_14 is hardcoded to be 0 instead of 1
```
    if (type & 0x80) {
        this->unk_14 = (type >> 5) & 0x1;
        this->typeColor = (type >> 3) & 0x3;
        this->speedType = type & 0x1;
        this->effect = (type >> 1) & 0x3;
    } else if (type == 1) {
        this->unk_14 = 1;
    } else {
        this->unk_14 = 0;
    }
    ```

Reply:
phrased another way, any transition with a type above 32 has its setType callback called twice in a row. And the circle only cares about what unk_14 is the first time its called. But in order for that first call to even happen it needs to be set...

Reply:
figured out and addressed, the parameter has been removed from the macro. unk_14 still has meaning, but not from the macro lol